src/main/scala/edu/ie3/simona/model/grid/GridModel.scala

```diff
@@ -407,7 +407,7 @@ object GridModel {
       noLines && noTransformers2w && noTransformers3w && (noOfNodes > noOfSlackNodes)
     )
       throw new InvalidGridException(
-        "The grid contains no basic branch elements (lines or transformers)."
+        f"The grid with subnet number ${gridModel.subnetNo} contains no basic branch elements (lines or transformers)."
```

Comment:
You could go one step further and add this: `Only grids that only consist of slack nodes are allowed to not contain lines and transformers` (or similar)

 (I had to deal with this not long ago and it took some time to understand the situation)

Reply:
Thanks, main issue was to get the subnet that raises the issue. Will adapt this, asking myself if it is longer valid with https://github.com/ie3-institute/simona/pull/1104

Reply:
That's a good question, but since you're only adapting the error message to describe current reality, this is fine for now imho